build-scripts/install-coverage.sh

```diff
@@ -3,13 +3,13 @@
 
 # This script should be executed only from meson as a part of install flow!
 #
+COVERAGE_ROOT="${MESON_INSTALL_DESTDIR_PREFIX}/share/bluechi-coverage"
+
 # Install all created `*.gcno` files so they could be packaged into bluechi-coverage RPM.
+mkdir -p ${COVERAGE_ROOT}
+( cd $MESON_BUILD_ROOT ; find . -name "*.gcno" -exec cp -v --parents {} ${COVERAGE_ROOT} \; )
 
-for f in $MESON_BUILD_ROOT/src/*; do
-    if [ -d "$f" ]; then
-        COV_DIR="${MESON_INSTALL_DESTDIR_PREFIX}/share/bluechi-coverage"
-        COV_DIR="${COV_DIR}/${f##*/}"
-        mkdir -p "${COV_DIR}"
-        find "${f}" -name '*.gcno' -exec cp "{}" "${COV_DIR}" \;
-    fi
-done
+# Unit test source files are not included in debugsource RPM, add them to bluechi-coverage RPM
```

Comment:
I'd deliberately not include the source files for the tests here. 
The unit tests are not part of the BlueChi "product", hence its not relevant which code was executing while running them. Only the resulting coverage of the actual source files, e.g. controller.c, are relevant. And these get collected without this block, right? So this can be removed.

Reply:
If you want to add unit test code coverage results, then you need to add also source files executing those tests to the report, otherwise you cannot create the report. For example if you want to add to code coverage report functions tested using https://github.com/eclipse-bluechi/bluechi/blob/main/src/libbluechi/test/common/cfg/cfg_def_conf_test.c then you need to add `cfg_def_conf_test.c` source file into sources, otherwise geninfo/genhtml will complain about missing sources

Reply:
I assume that the coverage information about these unit test source files is contained in the merged `.info` file. So a simple `lcov --remove <pattern> -o merged.info` should be enough. We can take care of this in a follow up PR, though.